app/models/course/assessment/submission.rb

```diff
@@ -135,6 +138,16 @@ def publish(_ = nil)
     end
   end
 
+  # Handles the unsubmission of a submitted submission.
+  #
+  # This will revert all manually graded answers to an attempting state.
```

Comment:
We need to discuss this again, but the `unsubmission` is primarily to: allow the student to edit their submission (Either they submit wrongly, or very bad submission - lots of errors). My assumption is that two things will happen: 
1. State of submission changes from `submitted`/`graded` to `attempting` - for abilities: student now can edit submission again.
2. Remove all awarded EXP. 

Minimally I try not to change the state of the question unless required. My assumption was that it is possible to re-attempt `auto_graded` questions - now that I think of it, is that correct? Might not be? 

The tradeoff is that if I unsubmit all questions, the student has to resubmit and re-evaluate all questions that are like `training` questions.